toutv/bos.py

```diff
@@ -58,9 +69,19 @@ def get_proxies(self):
 
     def _do_request(self, url, timeout=None, params=None):
         proxies = self.get_proxies()
+        auth = self.get_auth()
 
         try:
-            r = requests.get(url, params=params, headers=toutv.config.HEADERS,
+            headers = dict(toutv.config.HEADERS)
+
+            if auth and params:
```

Comment:
The only case where we don't have any params is when we download the segments from the Akamai server. I just wanted to avoid an uncessary HTTP request to get a claims.